File: 1450. Number of Students Doing Homework at a Given Time.rb
#* 1450. Number of Students Doing Homework at a Given Time
start_time = [1,2,3]; end_time = [3,2,7]; query_time = 4
# @param {Integer[]} start_time
# @param {Integer[]} end_time
# @param {Integer} query_time
# @return {Integer}
#*
def busy_student(start_time, end_time, query_time)
    start_time.zip(end_time).map{|a,b| (a..b).to_a}.count{|pair| pair.include?(query_time)}
end
p busy_student(start_time, end_time, query_time) 

#*
def busy_student(start_time, end_time, query_time)
    times = start_time.zip(end_time)
    times.map{|a,b| (a..b).to_a}.count{|pair| pair.include?(query_time)}
end
p busy_student(start_time, end_time, query_time) 

File: 1450. Number of Students Doing Homework at a Given Time.js
// 1450. Number of Students Doing Homework at a Given Time
startTime = [1,2,3]; endTime = [3,2,7];queryTime = 4
/**
 * @param {number[]} startTime
 * @param {number[]} endTime
 * @param {number} queryTime
 * @return {number}
 */
// SOL
var busyStudent = function(startTime, endTime, queryTime) {
    // Zip startTime and endTime into pairs
    const times = startTime.map((start, index) => [start, endTime[index]]);

    // Create ranges and count how many include queryTime
    const count = times
        .map(([a, b]) => Array.from({ length: b - a + 1 }, (_, i) => a + i))
        .filter(range => range.includes(queryTime))
        .length;

    return count;
}
console.log(busyStudent(startTime, endTime, queryTime))

// SOL
var busyStudent = function(startTime, endTime, queryTime) {
    const times = startTime.map((start, index) => [start, endTime[index]]);
    const count = times.filter(([a, b]) => queryTime >= a && queryTime <= b).length;

    return count;
}
console.log(busyStudent(startTime, endTime, queryTime))


